docs/badges/collecting.rst

```diff
@@ -0,0 +1,54 @@
+Collecting
+==========
+
+**Collecting** is a process of:
+
+- learners' `fulfillments` update subscription
+- badge requirements fulfillment/revocation analysis
+- completed badge awarding
+- badge revocation
+
+Fulfillments subscription
+-------------------------
+
+Configured `Badge Collector`_ is auto-subscribed for any updates in `Fulfillment`_ records.
+
+
+Badge completion analysis
+-------------------------
+
+- Once user's `Fulfillment`_ is updated, Collector tries to figure out if this progress update made the related Requirement to be **fulfilled**.
+- Collector emits `BADGE_REQUIREMENT_FULFILLED` signal.
+- If so, Collector goes further and checks all related requirements (if any).
+- If all requirements for given badge template are fulfilled, there are no obstacles to award the badge, so Collector emits internal `BADGE_REQUIREMENTS_COMPLETE` signal.
+
+
+Badge awarding
+--------------
+
+On `BADGE_REQUIREMENTS_COMPLETE` signal:
+
+- awarding handler creates new `UserCredential`_ record;
+- external event about a badge awarding fact is emitted;
+
+
+Badge revocation analysis
+-------------------------
+
+This is an alternative (reverse) pipeline step for requirements analysis with `revoke` effect set.
+
+- User's `Fulfillment`_ may be updated, so related Requirement stops being fulfilled.
+- Related badge template requirements become not fulfilled, so Collector emits internal `BADGE_REQUIREMENTS_NOT_COMPLETE` signal.
+
+Badge revocation
+----------------
+
+On `BADGE_REQUIREMENTS_NOT_COMPLETE` signal:
+
+- revocation handler updates `UserCredential`_ record's status to `revoked`;
```

Comment:
Will badges have proprietary statuses? Are they using the same `awarded` and `revoked` statuses that course and program credentials issued use?

Reply:
The `Badge` credential type is planned as the `UserCredential` descendant (so, internal statuses will be consistent). There will be a special extended type specific for Credly though, but it will live in the Credly "backend" plugin.
Generally, we want to split "core", simple, self-contained badges functionality, and Credly-specific extensions.